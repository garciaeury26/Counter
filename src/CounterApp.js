import React, { useState } from 'react'
import { useCounter } from './hooks/useCounter'
import { Reset } from './Reset'

const h1 = <h1>React is amazing!</h1>

const CounterApp = ({ value }) => {

    const { number, sumar, restar, reset } = useCounter(value);

    return (
        <div className="container">
            <header className="header">
                {h1}
            </header>
            <h2>{number}</h2>
            <div className='conter-div'>
                <button onClick={sumar} type="button">Sumar</button>
                <button onClick={restar} type="button">Restar</button>
                <button onClick={reset} type="button">Reset</button>
                {/* <Reset setNumber={setNumber} /> */}
            </div>
        </div>
    )
}

CounterApp.defaultProps = {
    subtitulo: 'vaca: klk mio tine problema en los ri√±ones'
};

export default CounterApp
