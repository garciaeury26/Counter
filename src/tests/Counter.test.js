
import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from 'enzyme'

import CounterApp from '../CounterApp';


describe('Prebas en el componente Counter app', () => {

    let wrapper = shallow(<CounterApp value={100} />)

    //reiniciar el valor cada prueba
    beforeEach(() => {
        wrapper = shallow(<CounterApp value={100} />)
    })

    test('Deve de mostrar <CounterApp/> correctamente (hacer macht con un snapshot)', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Deve de mostrar el valor por defecto de 100', () => {
        const value = wrapper.find('h2').text();
        expect(value).toBe('100');
    });

    test('Deve de incrementar el valor +1', () => {

        //el primer boton seria el que esta en el indice cero
        wrapper.find('button').at(0).simulate('click');
        const value = wrapper.find('h2').text();

        expect(value).toBe('101');
    });

    test('Deve de decrementar el valor -1', () => {
        wrapper.find('button').at(1).simulate('click');
        const value = wrapper.find('h2').text();

        expect(value).toBe('99')
    })

    test('El valor tiene que volver a suvalor por defecto', () => {
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click');
        const value = wrapper.find('h2').text();

        expect(value).toBe('100')
    })
});


//! ojo las pruebas se realizan de forma sicrona entonces si yo tengo declarada un variable declarada de forma global con el componente
//! si simulo un +1 a una prop el valor de la variable cambiara
//* beforeEach arregla es
