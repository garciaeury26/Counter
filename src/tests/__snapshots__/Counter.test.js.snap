// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prebas en el componente Counter app Deve de mostrar <CounterApp/> correctamente (hacer macht con un snapshot) 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <h1>
      React is amazing!
    </h1>
  </header>
  <h2>
    100
  </h2>
  <div
    className="conter-div"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Sumar
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Restar
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;
